### ISSUE TITLE
Commerce Settings

### ISSUE THREAD

**Issue by jaymarvels** [Geta/geta-optimizely-sitemaps#93]
Are there any additional configuration to get this to work with commerce apart from adding:

services.AddSitemapsCommerce();

In the configuration docs (https://github.com/Geta/geta-optimizely-sitemaps/blob/master/docs/editor-guide.md) it talks about the format and how its currently standard or mobile. If I do check the "standard and commerce" option and run the scheduled job my commerce product pages still do not show in the sitemap.xml.

Is something missing?

**Comment by jaymarvels** [comment]
@marisks Any input?

**Comment by marisks** [comment]
As I remember, it is enough to follow configuration documentation here: https://github.com/Geta/geta-optimizely-sitemaps/tree/master#configuration
So call AddSitemaps, AddSitemapsCommerce, and make sure that razor pages are mapped.

**Comment by jaymarvels** [comment]
@marisks Yeah unfortunately its not working. I am guessing it could be due to it being multi site, multi language, multi catalog it can't get the correct one.

If I create my own sitemap and commerce generator by implementing the public virtual member, eg:

```
public class StandardAndCommerceSitemapXmlGenerator : CommerceSitemapXmlGenerator, ICommerceAndStandardSitemapXmlGenerator
{
//
}
```

And just use the default from the repo as an override:

```
        protected override IEnumerable<XElement> GetSitemapXmlElements()
        {
            var rootContentReference = _referenceConverter.GetRootLink();

            if (SitemapData.RootPageId != Constants.DefaultRootPageId)
            {
                rootContentReference = new ContentReference(SitemapData.RootPageId)
                {
                    ProviderName = "CatalogContent"
                };
            }

            var descendants = ContentRepository.GetDescendents(rootContentReference);

            return GenerateXmlElements(descendants);
        }
```
Then descendants is always empty.

**Comment by marisks** [comment]
Have you configured RootPageId for the catalog you want to generate a sitemap?

**Comment by alex-finnigan-mando** [comment]
Bumping this issue as I've also run into the same problem but I think I've found the cause.

This bug only manifests when the user sets the Root page ID to anything other than -1 (which could be the case in a multisite scenario). 

In `CommerceAndStandardSitemapXmlGenerator`, the `GetSitemapXmlElements` method gets the CMS content then calls the base method from `CommerceSitemapXmlGenerator`. In this base method, we get the Catalog Root using `ReferenceConverter` and then the problem occurs in the following IF statement:
```
if (SitemapData.RootPageId != Constants.DefaultRootPageId)
{
    rootContentReference = new ContentReference(SitemapData.RootPageId)
    {
        ProviderName = "CatalogContent"
    };
}
```
this will be evalutated as true since we have defined a specific page instead of the default root. This will then override the Catalog Root to -1, and since -1 does not exist in the Commerce tree, we get no descendants.

I assume the fix for this would be to remove this IF statement because I can't see why we would need to do this?

Current workaround for anyone searching for answers would be to follow @jaymarvels' lead and override the `CommerceAndStandardSitemapXmlGenerator` service with a copy in your solution, then remove the IF statement (unless there's a good reason why we shouldn't?).
